### ISSUE TITLE
"100 Above" callout during CAT III Autoland with no decision height

### ISSUE THREAD

**Issue by groundspoilers** [flybywiresim/a32nx#6504]
### Aircraft Version

Development

### Build info

```json
{
  "built": "2022-01-04T06:33:26+00:00",
  "ref": "master",
  "sha": "93b60b06f5db427fe1d9bf93531a513a3562845f",
  "actor": "tracernz",
  "event_name": "manual"
}
```

### Describe the bug

When programming a CAT 3 Dual autoland for Glasgow Airport (EGPF), I entered "NO" into the "radio" entry on the APPR Phase page. This then appeared on the FMA as "No DH". During the landing at approximately 100ft AGL, the bug occurred. The aircraft called out "100 Above" indicating that I was 100 feet above minimums. This is not meant to happen as there were no minimums selected for the approach.

### Expected behavior

I expected there to be no "100 Above" callout as no minimums were selected for the autoland.

### Steps to reproduce

1. Fly the RIBEL1G STAR into runway 23 at EGPF.
2. Set up the aircraft for a CAT 3 Dual Autoland with no decision height.
3. Allow the aircraft to land and listen for a "100 Above" callout

### References (optional)

https://user-images.githubusercontent.com/97336141/148641035-9790eaa5-b24c-4bb3-87d2-56b5641ff203.mp4

### Additional info (optional)

This bug hasn't happened in previous autoland procedures with no decision height. This is my first time encountering this issue.

### Discord Username (optional)

UnstableExplosives#4043

**Comment by tshomas** [comment]
Asked to create issue report by myself, apologies if issue has been made before but couldn’t find anything.

![7A1D7A8B-BF22-4BDE-90CF-73B02B374A4A](https://user-images.githubusercontent.com/74841560/148641166-d5221052-5e40-4439-ad8a-2230058862d7.jpeg)

**Comment by aguther** [comment]
Is that really related to Autoland? My guess is that it’s related to FWC in general what it does when no decision height is entered.

What happens in manual landing case and what should happen?

**Comment by tshomas** [comment]
Yeah it’s probably more FWC but connected to autoland by the fact of that’s the only case where it is not heard. 

CAT 3 you’d enter NO or NO DH which would stop the “100 above” callout. 
For every other landing with a minima then you’d hear it.

Either way it’s a bug, probably to be corrected through the FWC.

**Comment by aguther** [comment]
What I mean is what is IRL behavior when you do not enter a minima and you fly manually?

**Comment by tshomas** [comment]
Now not a situation I know amazingly well as I put a minima for most things, but if both fields are left blank then there are no auto call-outs for the “100 above” or “minimum”. 

The notes I have for my INN training say that as RNAV 08 has a minima that is too high for the FMGCs to always accept, then to expect no call-outs on the approach. 

This is only anecdotal without full proof, but I’m on INN training this week so may get to confirm it. It’s also what I would expect as I wouldn’t expect a callout with neither field filled in.

I believe as well, once again without testing, that the A32NX models this as I’ve seen enough user reports saying why they don’t get a minima callout but it turns out they never filled in a field. Would need to test by someone with some spare time.

However not sure what that has to do with the issue above. The “NO” or “NO DH” setting in RADIO is causing the “100 Above” to sound at 100 AMSL, as seen in the video, which is incorrect behaviour.

**Comment by aguther** [comment]
So from what I see now is that when you enter a DH of 0 then the sound needs to be inhibited beside the case when no DH is entered at all.

Correct?

**Comment by tshomas** [comment]
So if “NO” or “NO DH” or “NODH” is entered into RADIO then both “minimum” and “100 above” should be muted as there is no minima to base them off.

I cannot find if “0” in RADIO also does the same in FCOM, we may need to check a Honeywell manual.

**Comment by tracernz** [comment]
Entering NO NODH or NO DH gives a special value of -2. Otherwise anything from 0 to 5000 inclusive will be accepted as a normal DH https://github.com/flybywiresim/a32nx/blob/master/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/FMC/A32NX_FMCMainDisplay.js#L3200.

The FMA correctly picks up the special value and displays "NODH" https://github.com/flybywiresim/a32nx/blob/e7bdd9796930970cbce0267a894bfff052d9f714/src/instruments/src/PFD/FMA.tsx#L693.

The GPWS module that generates the callout does not pick up -2 correctly https://github.com/flybywiresim/a32nx/blob/a9236d7d62ec9bda460c4751278bac5fce844d13/flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_GPWS.js#L140. That is the bug.

**Comment by beheh** [comment]
While one fix would be to apply the suggested patch, this should also be fixed once the new FWC over at #4872 lands.

**Comment by tracernz** [comment]
I posted it in one of the discord channels hoping to bring on a new contributor with a nice simple patch but no bites. 😨

**Comment by patmack14** [comment]
Id be willing to look into this if it is still available.

**Comment by sidnov** [comment]
> Id be willing to look into this if it is still available.
> 

It will be fixed with the new FWC, Beheh is already working on it

**Comment by tracernz** [comment]
> Id be willing to look into this if it is still available.

Yes please. very simple patch per https://github.com/flybywiresim/a32nx/issues/6504#issuecomment-1008344352 that will do no harm until the new FWC is ready.

**Comment by patmack14** [comment]
@tracernz sorry for delay here, is this STILL available lol?  
Ready to start working on if #4872 is still a ways out

**Comment by tracernz** [comment]
> @tracernz sorry for delay here, is this STILL available lol? Ready to start working on if #4872 is still a ways out

Yep, go ahead :D.
